lib/vintage_net_wizard/web/endpoint.ex

```diff
@@ -58,33 +61,27 @@ defmodule VintageNetWizard.Web.Endpoint do
     ]
   end
 
-  defp maybe_use_ssl(_use_ssl = true) do
-    ssl_dir = ssl_dir()
+  defp maybe_use_ssl(_use_ssl = true, opts) do
+    port = Application.get_env(:vintage_net_wizard, :port, 443)
+    ssl_options = Keyword.get(opts, :ssl)
+    options = [dispatch: dispatch(), port: port]
 
     Plug.Cowboy.child_spec(
       plug: Router,
       scheme: :https,
-      options: [
-        dispatch: dispatch(),
-        certfile: Application.get_env(:vintage_net_wizard, :certfile, "#{ssl_dir}/cert.pem"),
-        keyfile: Application.get_env(:vintage_net_wizard, :keyfile, "#{ssl_dir}/key.pem"),
-        port: Application.get_env(:vintage_net_wizard, :port, 443)
-      ]
-    )
-  end
-
-  defp maybe_use_ssl(_no_ssl) do
-    Plug.Cowboy.child_spec(
-      plug: Router,
-      scheme: :http,
-      options: [
-        dispatch: dispatch(),
-        port: Application.get_env(:vintage_net_wizard, :port, 80)
-      ]
+      options: Keyword.merge(ssl_options, options)
     )
   end
 
-  defp ssl_dir() do
-    Path.join(:code.priv_dir(:vintage_net_wizard), "ssl")
+  defp maybe_use_ssl(_no_ssl, _opts) do
+    {:ok,
+     Plug.Cowboy.child_spec(
+       plug: Router,
+       scheme: :http,
+       options: [
+         dispatch: dispatch(),
+         port: Application.get_env(:vintage_net_wizard, :port, 443)
```

Comment:
```suggestion
         port: Application.get_env(:vintage_net_wizard, :port, 80)
```
This is the _no ssl_ option, so it should default to port 80 for `http`